lib/kernel.ts

```diff
@@ -796,6 +956,52 @@ export class Kernel {
 			);
 		}
 
+		// If inserting a link, validate that there's a relationship for it
+		const contractType = contract.type.split('@')[0];
+		if (contractType === 'link') {
+			context.debug('Fetching candidate relationships for link', {
+				slug: contract.slug,
+			});
+			const linkContract: LinkContract = contract as LinkContract;
+			const unversionedFromType = linkContract.data.from.type.split('@')[0];
+			const unversionedToType = linkContract.data.to.type.split('@')[0];
+
+			const results = this.getRelationships().filter(
+				(rel) =>
+					rel.type === 'relationship@1.0.0' &&
+					// forward relationship
+					((linkContract.name === rel.name &&
+						linkContract.data.inverseName === rel.data.inverseName &&
+						[linkContract.data.from.type, unversionedFromType, '*'].includes(
+							rel.data.from.type,
+						) &&
+						[linkContract.data.to.type, unversionedToType, '*'].includes(
+							rel.data.to.type,
+						)) ||
+						// inverse relationship
+						(linkContract.data.inverseName === rel.name &&
+							linkContract.name === rel.data.inverseName &&
+							[linkContract.data.from.type, unversionedFromType, '*'].includes(
+								rel.data.to.type,
+							) &&
+							[linkContract.data.to.type, unversionedToType, '*'].includes(
+								rel.data.from.type,
+							))),
+			);
+
+			if (results.length === 0) {
+				const errorMessage = `Relationship not found ${
+					linkContract.data.from.type
+				} ${linkContract.name} ${
+					linkContract.data.to.type
+				} for contract ${JSON.stringify(contract)}`;
```

Comment:
Can you change this to just use the contract ID here? It will reduce the amount of logging and also prevent situations where we might be leaking private data in logs (something I want to avoid as much as possible)

Reply:
Fixed. It was useful while developing but yes, it's too much.